libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1596,7 +1595,7 @@ bool TypeChecker::visit(TupleExpression const& _tuple)
 				if (dynamic_cast<TupleType const&>(*types[i]).components().empty())
 				{
 					if (_tuple.isInlineArray())
-						m_errorReporter.fatalTypeError(5604_error, components[i]->location(), "Array component cannot be empty.");
+						m_errorReporter.typeError(5604_error, components[i]->location(), "Array component cannot be empty.");
```

Comment:
Why not fatal anymore?

Reply:
With not fatal error here, the compiler can output more detailed error messages for some cases e.g. for 
```
contract C {
        function f() private pure {}
        function a() public {
            uint x;
            uint y;
            (x, y) = [f(), f()];
        }
    }
```

the compiler outputs 
```
TypeError 5604: (122-125): Array component cannot be empty.
TypeError 9656: (121-131): Unable to deduce nameable type for array elements. Try adding explicit type conversion for the first element.
```
instead of just: 
`TypeError 5604: (122-125): Array component cannot be empty.`

I know this is not a perfect error message in that case. But still, an extra error message helps to reroute user attention to the real root cause direction.

Reply:
Well, the more I think about it, the more I believe the second error is misleading in this case. At the beginning I thought maybe it can direct user eyes to the return type of the "f()" function. However, the suggestion to add explicit type conversion is absolutely misguided.   Simultaneously, information abut attempt to create an array with empty components should be enough. So that, I will make the 5604 fatal again.

Reply:
Yes, sounds good!